tutorial/creating_packages/build_packages.rst

```diff
@@ -1,2 +1,220 @@
+  some unit tests over this function. This function is just creating an output message
+  based on the arguments passed.
+
+* The `CMakeLists.txt
+  <https://github.com/conan-io/libhello/blob/with_tests/CMakeLists.txt#L15-L17>`_ uses a
+  ``BUILD_TEST`` CMake variable that conditionally adds the *tests* directory.
+
+.. code-block:: cmake
+    :emphasize-lines: 12, 19, 27-28, 35-36
+
+    cmake_minimum_required(VERSION 3.15)
+    project(hello CXX)
+
+    ...
+
+    if (BUILD_TESTS)
+        add_subdirectory(tests)
+    endif()
+
+    ...
+
+* We have a `CMakeLists.txt
+  <https://github.com/conan-io/libhello/blob/with_tests/tests/CMakeLists.txt>`_ in the
+  *tests* folder using `googletest <https://github.com/google/googletest>`_ for
+  testing.
+
+.. code-block:: cmake
+
+    cmake_minimum_required(VERSION 3.15)
+    project(PackageTest CXX)
+
+    find_package(GTest REQUIRED CONFIG)
+
+    add_executable(test_hello test.cpp)
+    target_link_libraries(test_hello GTest::gtest GTest::gtest_main hello)
+
+
+With basic tests on the functionality of the ``compose_message()`` function:
+
+
+.. code-block:: cpp
+
+    #include "../include/hello.h"
+    #include "gtest/gtest.h"
+
+    namespace {
+        TEST(HelloTest, ComposeMessages) {
+        EXPECT_EQ(std::string("hello/1.0: Hello World Release! (with color!)\n"), compose_message("Release", "with color!"));
+        ...
+        }
+    }
+
+
+Changes introduced in the recipe
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* We added the *gtest/1.11.0* requirement to the recipe as a ``test_requires()``. This is
+  a special type of requirement intended for testing libraries like **Catch2** or **gtest**.
+
+* We use the ``tools.build:skip_test`` configuration, to tell CMake wether to build and
+  run the tests or not. This configuration controls the execution of ``CMake.test()`` and
+  ``Meson.test()`` but can also be used for other testing environments like in this case.
+  We use that variable in the ``generate()`` method to inject the ``BUILD_TESTS`` variable
+  to CMake and also in the ``build()`` method after building the package and the tests to
+  run the tests.
+
+
+Conditionally patching the source code
+--------------------------------------
+
+You can also use the ``build()`` method to apply patches to the source code before
+launching the build based on the value of settings or options. There are several ways to
+do this in Conan. One of them would be using the :ref:`replace_in_file
+<conan_tools_files_replace_in_file>` tool:
+
+.. code-block:: python
+
+    import os
+    from conan import ConanFile
+    from conan.tools.files import replace_in_file
+
+
+    class helloRecipe(ConanFile):
+        name = "hello"
+        version = "1.0"
+
+        # Binary configuration
+        settings = "os", "compiler", "build_type", "arch"
+        options = {"shared": [True, False], "fPIC": [True, False]}
+        default_options = {"shared": False, "fPIC": True}
+
+        def build(self):
```

Comment:
Add a note/warning, saying that patching in build() should only be done for very special cases that the code will be different, but clarify that it will make more challenging the local flow with ``conan install`` + native build, and probably for those cases it is better to try a ``conan build`` command that patches for you locally too.